### ISSUE TITLE
Inconsistent test results with crop_size

### ISSUE THREAD

**Issue by thocevar** [BVLC/caffe#1767]
Using `crop_size` leads to different test results reported during the training/testing process and in a separate testing run of the caffe tool.

As an example consider adding `crop_size: 24` to the transformation parameters of the data layers in the training and testing phase of the LeNet network example for the MNIST dataset (`examples/mnist/lenet_train_test.prototxt`). Training the model with intermediate testing every 100 iterations (`./build/tools/caffe train --solver=examples/mnist/lenet_solver.prototxt`) reports
> Snapshotting to examples/mnist/lenet_iter_100.caffemodel
> Snapshotting solver state to examples/mnist/lenet_iter_100.solverstate
> Iteration 100, Testing net (#0)
>     Test net output # 0: accuracy = 0.7269
>     Test net output # 1: loss = **0.88038** (* 1 = 0.88038 loss)

However, following with a testing run (`./build/tools/caffe test --model=examples/mnist/lenet_train_test.prototxt --weights=examples/mnist/lenet_iter_100.caffemodel --iterations=100`) gives
> accuracy = 0.7868
> loss = **0.71078** (* 1 = 0.71078 loss)

The results are the same if you don't set `crop_size`.

**Comment by shelhamer** [comment]
Duplicate of #1430 -- which needs to be fixed.
